### DISCUSSION TITLE
After uninstalling, all the terminal history is gone, npm, nvm are back to very old versions

### DISCUSSION THREAD

**Discussion by CharlesJQuarra** [ohmyzsh/ohmyzsh#9890] [Help] [answered]
After trying several days of failing miserably to tweak configurations options to avoid terminal history being shared across iTerm tabs with 
```
setopt share_history Off 
setopt share_history 0
```

I uninstalled oh-my-zsh, but my terminal history is now all gone. Also npm, nvm and lots of other command line apps I had are now inaccesible (or point to very old versions) and is not easy to figure out where each of them might be installed and fix it.

I tried installing again `oh-my-zsh`, and the terminal history is available again, BUT npm, nvm, node, etc. are still pointing to the very old versions

---
## EDIT

Ok I recovered npm/nvm by adding `export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm` inside `~/.oh-my-zsh/oh-my-zsh.sh` which allows me to remain productive while I figure out either how to disable this annoying sharing of command history across tabs, or able to export my terminal history before uninstalling oh-my-zsh

**Comment by mcornella** [comment, marked as answer]
> After trying several days of failing miserably to tweak configurations options to avoid terminal history being shared across iTerm tabs with
> 
> ```
> setopt share_history Off 
> setopt share_history 0
> ```

The right command is `unsetopt share_history`
